### PR TITLE
Test overriding the noobaa-default backingstore + add support for RGW backingstore creation

### PR DESCRIPTION
**Pull request by sagihirshfeld**
Coverage for https://url.corp.redhat.com/140509f

The implementation of the upgrade TC required to override the default backingstore with a backingstore of the same type, in order to limit the affect on other tests. To handle the cases where the default backingstore is using RGW (like in our vSphere deployments) support for RGW backingstore creation has been added, and two basic test cases has been added to validate this support while extending our coverage:

- tests/manage/mcg/test_write_to_bucket.py::TestBucketIO::test_write_file_to_bucket[RGW-OC-1]
- tests/manage/mcg/test_write_to_bucket.py::TestBucketIO::test_write_file_to_bucket[RGW-CLI-1]

Requires a backport to release-4.14

### CODE REVIEW

**Review by keesturam — APPROVED** [review]
neat PR! 
LGTM.

File: tests/functional/object/mcg/test_default_backingstore_override.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+        }
+        replication_policy_patch_dict = {
+            "spec": {"replicationPolicy": json.dumps(replication_policy)}
+        }
+        OCP(
+            kind=constants.BUCKETCLASS,
+            namespace=config.ENV_DATA["cluster_namespace"],
+            resource_name=constants.DEFAULT_NOOBAA_BUCKETCLASS,
+        ).patch(params=json.dumps(replication_policy_patch_dict), format_type="merge")
+
```
Comment (udaysk23) [inline tests/functional/object/mcg/test_default_backingstore_override.py:160]: Please use `patch_replication_policy_to_bucketclass` method here which is present in ocs_ci/ocs/bucket_utils.py

Comment (sagihirshfeld) [reply tests/functional/object/mcg/test_default_backingstore_override.py]: Thanks for pointing this out! That method was created with an OBC as the target in mind so I made the needed adjustments to use it to patch a bucketclass as well in order to use in `test_bucketclass_replication_after_default_backingstore_override`

**Review by mashetty330 — COMMENTED** [review]
LGTM, this may need validation of both MCG tier1, tier2 tests. please run it.

**Review by mashetty330 — APPROVED** [review]

**Review by udaysk23 — APPROVED** [review]
LGTM

**Review by keesturam — APPROVED** [review]

**Pull request merged**